packages/victory-bar/src/bar.js

```diff
@@ -53,17 +55,141 @@ export default class Bar extends React.Component {
+      const topLine = `L ${x1 - cornerRadius.top}, ${y1}`;
+      const topRightArc = `A ${topArc}, ${x1}, ${y1 + sign * cornerRadius.top}`;
+      const rightLine = `L ${x1}, ${y0 - sign * cornerRadius.bottom}`;
+      const bottomRightArc = `A ${bottomArc}, ${x1 - cornerRadius.bottom}, ${y0}`;
+      const end = "z";
+
+      return [ start,
+        bottomLeftArc,
+        leftLine,
+        topLeftArc,
+        topLine,
+        topRightArc,
+        rightLine,
+        bottomRightArc,
+        end ].join("\n");
+    }
+
+    // There are three ways the verticalBarPath can self-intersect
+    // 1.) The top and bottom cornerRadius are both > 0 (the topArc and bottomArc intersect)
+    // 2.) The top cornerRadius > 0 but the bottom cornerRadius === 0 (the topArc and bottomLine intersect)
+    // 3.) The top cornerRadius === 0 but the bottom cornerRadius > 0 (the bottomArc and topLine intersect)
+
+    // case 1
+    if (cornerRadius.top !== 0 && cornerRadius.bottom !== 0) {
+
+      // find intersection of top left and bottom left arc
+      const topLeftCenter = new Point(x0 + cornerRadius.top, y1 + sign * cornerRadius.top);
+      const topLeftCircle = new Circle(topLeftCenter, cornerRadius.top);
+
+      const bottomLeftCenter = new Point(x0 + cornerRadius.bottom, y0 - sign * cornerRadius.bottom);
+      const bottomLeftCircle = new Circle(bottomLeftCenter, cornerRadius.bottom);
+
+      // circles intersect at two points, find the correct one
+      const intrxnsLeft = topLeftCircle.intersection(bottomLeftCircle);
+      const intrxnLeft = intrxnsLeft.length === 1 ? intrxnsLeft[0] :
+        intrxnsLeft[0].x <= intrxnsLeft[1].x ? intrxnsLeft[0] :
+        intrxnsLeft[1];
+
+      // find intersection of top right and bottom right arc
+      const topRightCenter = new Point(x1 - cornerRadius.top, y1 + sign * cornerRadius.top);
+      const topRightCircle = new Circle(topRightCenter, cornerRadius.top);
+
+      const bottomRightCenter = new Point(x1 - cornerRadius.bottom, y0 - sign * cornerRadius.bottom);
+      const bottomRightCircle = new Circle(bottomRightCenter, cornerRadius.bottom);
+
+      // circles intersect at two points, find the correct one
+      const intrxnsRight = topRightCircle.intersection(bottomRightCircle);
+      const intrxnRight = intrxnsRight.length === 1 ? intrxnsRight[0] :
+        intrxnsRight[0].x >= intrxnsRight[1].x ? intrxnsRight[0] :
+        intrxnsRight[1];
+
+      const start = `M ${x0 + cornerRadius.bottom}, ${y0}`;
+      const bottomLeftArc = `A ${bottomArc}, ${intrxnLeft.x}, ${intrxnLeft.y}`;
+      const topLeftArc = `A ${topArc}, ${x0 + cornerRadius.top}, ${y1}`;
+      const topLine = `L ${x1 - cornerRadius.top}, ${y1}`;
+      const topRightArc = `A ${topArc}, ${intrxnRight.x}, ${intrxnRight.y}`;
+      const bottomRightArc = `A ${bottomArc}, ${x1 - cornerRadius.bottom}, ${y0}`;
+      const end = "z";
+
+      return [ start,
```

Comment:
You can just return once at the end after modifying any values you need to modify for intersections. It also wont alter the final path to leave in zero radius arcs, for example.